### ISSUE TITLE
azure - container-host - c7n.exceptions.InvalidOutputConfig

### ISSUE THREAD

**Issue by bramvdklinkenberg** [cloud-custodian/cloud-custodian#5811]
**Describe the bug**
I followed this [aci tutorial](https://cloudcustodian.io/docs/azure/configuration/acitutorial.html#azure-configuration-acitutorial), but I keep getting the error:
`c7n.exceptions.InvalidOutputConfig: Invalid c7n.output.blob: https://<sa-account-name>.blob.core.windows.net/c7n-aci-logs`

**To Reproduce**
Steps to reproduce the behavior: See [tutorial](https://cloudcustodian.io/docs/azure/configuration/acitutorial.html#azure-configuration-acitutorial)

**Expected behavior**
The aci tutorial to work

**Background (please complete the following information):**
 - OS: N/A
 - Python Version: N/A
 - Custodian Version: latest image tag
 - Tool Version: [if applicable]
 - Cloud Provider: azure
 - Policy: [please exclude any account/sensitive information]
policies:
```
- name: find-c7nstorage
  resource: azure.storage
  mode:
    type: container-periodic
    schedule: "* * * * *"  # Run every minute as an example
  filters:
  - type: value
    key: name
    op: eq
    value: <sa-account-name>
```
policies: 
   - name: find-c7nstorage
      resource: azure storage account
```

 - Traceback: [if applicable, please exclude sensitive/account information]
```
Traceback (most recent call last):
  File "/src/tools/c7n_azure/c7n_azure/container_host/host.py", line 408, in run_policy
    policy.push(event, context)
  File "/src/c7n/policy.py", line 1126, in push
    return mode.run(event, lambda_ctx)
  File "/src/tools/c7n_azure/c7n_azure/container_host/modes.py", line 70, in run
    return PullMode.run(self)
  File "/src/c7n/policy.py", line 282, in run
    with self.policy.ctx:
  File "/src/c7n/ctx.py", line 83, in __enter__
    self.initialize()
  File "/src/c7n/ctx.py", line 57, in initialize
    self.output = blob_outputs.select(self.options.output_dir, self)
  File "/src/c7n/output.py", line 62, in select
    raise InvalidOutputConfig("Invalid %s: %s" % (
c7n.exceptions.InvalidOutputConfig: Invalid c7n.output.blob: https://sa-account-name.blob.core.windows.net/c7n-aci-logs
```

**Comment by kapilt** [comment]
seems like this is missing the azure:// prefix on the storage blob

**Comment by bramvdklinkenberg** [comment]
```
        "azure_container_policy_uri": {
            "value": "<Policy Storage Container URI>"
        },
        "azure_container_storage_resource_id": {
            "value": "<Storage Account Resource ID>"
        },
        "azure_container_log_group": {
            "value": "azure://<Instrumentation Key>"
        },
        "azure_container_metrics": {
            "value": "azure://<Instrumentation Key>"
        },
        "azure_container_output_dir": {
            "value": "<Log Storage Container URI>"
        }
```

In the tutorial they only use that for the instrumentation key of the insights instance.

**Comment by nancybelser** [comment]
I am getting this same error. I followed the tutorial instructions as mentioned in the comment above.

**Comment by kapilt** [comment]
I'll try and loop back around to this after finishing the azure functions managed identity pr #5867

**Comment by kapilt** [comment]
per the tutorial you should be prefixing storage urls with azure:// ie azure_container_output_dir which neither of your examples seem to be doing afaics. 

ie from the tutorial.

```
AZURE_CONTAINER_LOG_GROUP |   | The application insights to send log output to. In the format: azure://<instrumentation_key_guid>.
AZURE_CONTAINER_METRICS |   | The application insights to send metrics output to. In the format: azure://<instrumentation_key_guid>.
AZURE_CONTAINER_OUTPUT_DIR |   | The URL of the storage account blob container to send log output to. In the format: azure://<storage_account_name>.blob.core.windows.net/<blob_container_name>.

```

I'm not huge fan of the container host mode as imo it doesn't represent best practices wrt to custodian setups imo as it ends up running a scheduler in python and consuming policies from an always running container instead of deferring to native tools that handle scheduling better. That said there are users using it successfully. imo custodian best practice starts with doing a policy as code setup around a git repository with ci around changes and deferring to other tools for scheduling (cron, container cloud periodic timers, etc). the two potentially can be combined, but its better to start with policy as code and a gitops setup, as that ensures you have audit and review on policy changes. part of the historical reason for container mode afaict was issues with azure functions, but those are resolved for event / realtime executions. else just running c7n-org in a cron will allow multi-sub execution, can can be run from any compute infra (container, Jenkins, etc).

**Comment by nancybelser** [comment]
Hi Kapil,

I made that change and reran and it worked.  Thanks.

I would like to point out that there are inconsistencies in the documentation on the cloudcustodian site.  Screenshot below is from the helm tutorial page https://cloudcustodian.io/docs/azure/configuration/helmtutorial.html
That page shows the container output dir as using the same format for as the container policy uri.  I originally used the same format for both.  I am a little confused as to how cloudcustodian can read the policies from the policy URI w/o ‘azure://’ in front of it, but the same format does not work for the output dir.  Is there a particular reason for the format difference?

[cid:image001.png@01D64626.FF50DF40]

As for your points about using container host mode, this is just a proof of concept.  We run everything in containers in our environment and already have an existing AKS cluster for our tools so I deployed to that cluster.  I converted the ‘az’ commands to terraform as per our internal standards and long term will be deploying using our existing github actions CD pipeline.  I would prefer not to have an always running container, so I will look into other options if we decide to use cloudcustodian long term.
